### PR TITLE
Log the error when unmarshal task from acs failed

### PR DESCRIPTION
**Pull request by richardpen**
### Testing

- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: agent/acs/handler/payload_handler.go [hunk 1]
```diff
@@ -325,6 +325,8 @@ func isTaskStatusNotStopped(status api.TaskStatus) bool {
 // handleUnrecognizedTask handles unrecognized tasks by sending 'stopped' with
 // a suitable reason to the backend
 func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs.Task, err error, payload *ecsacs.PayloadMessage) {
+	seelog.Warnf("Received unexpected acs message, messageID: %s, task: %v, err: %v", *payload.MessageId, task, err)
```
Comment (aaithal) [inline agent/acs/handler/payload_handler.go]: please use aws.StringValue for printing message id. Also, let's not print the entire task here. Can you just print just the arn instead?

Comment (sharanyad) [reply agent/acs/handler/payload_handler.go]: I think task arn can be nil here.

Comment (aaithal) [reply agent/acs/handler/payload_handler.go]: `aws.StringValue(task.Arn)` should cover that.

File: agent/acs/handler/payload_handler.go [hunk 2]
```diff
@@ -325,8 +325,11 @@ func isTaskStatusNotStopped(status api.TaskStatus) bool {
 // handleUnrecognizedTask handles unrecognized tasks by sending 'stopped' with
 // a suitable reason to the backend
 func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs.Task, err error, payload *ecsacs.PayloadMessage) {
+	seelog.Warnf("Received unexpected acs message, messageID: %s, task: %v, err: %v",
+		aws.StringValue(payload.MessageId), aws.StringValue(task.Arn), err)
+
 	if task.Arn == nil {
```
Comment (aaithal) [inline agent/acs/handler/payload_handler.go]: Let's also modify this to be `if aws.StringValue(task.Arn) == ""`. That is a more complete check, that'll check both for the field not being set (`nil`) and set to an empty string.

**Comment by richardpen** [comment]
@aaithal Removed the `amazon-ecs-cni-plugins` from this PR.

File: agent/acs/handler/payload_handler.go [hunk 3]
```diff
@@ -325,8 +325,11 @@ func isTaskStatusNotStopped(status api.TaskStatus) bool {
 // handleUnrecognizedTask handles unrecognized tasks by sending 'stopped' with
 // a suitable reason to the backend
 func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs.Task, err error, payload *ecsacs.PayloadMessage) {
-	if task.Arn == nil {
-		seelog.Criticalf("Received task with no arn, messageId: %s, task: %v", *payload.MessageId, task)
+	seelog.Warnf("Received unexpected acs message, messageID: %s, task: %v, err: %v",
+		aws.StringValue(payload.MessageId), aws.StringValue(task.Arn), err)
+
+	if aws.StringValue(task.Arn) == "" {
+		seelog.Criticalf("Received task with no arn, messageId: %s, task: %v", aws.StringValue(payload.MessageId), aws.StringValue(task.Arn))
```
Comment (aaithal) [inline agent/acs/handler/payload_handler.go]: nit: we don't need to print `task.Arn` here since we know it's not set.

**Pull request merged**